backend/translation/views.py

```diff
@@ -1,199 +1,246 @@
         try:
             translation = Translation.objects.get(
-                pk=translation_id, target_lang=target_lang
+                pk=translation_id, target_language=target_language
             )
-            # If the translation mentioned does not belong to the current user,
-            # create a new translation with parent as referred translation_id
-            if translation.user != user:
-                new_translation = Translation.objects.create(
-                    translation_type="mc",
-                    parent=translation,
-                    transcript=translation.transcript,
-                    target_lang=target_lang,
-                    user=user,
-                    payload=captions,
-                )
-                new_translation.save()
-                created = True
-            # Update the existing translation
-            else:
-                translation.payload = captions
-                translation.translation_type = "he"
-                translation.save()
-        # If no translation exists for the given translation_id and target_lang,
-        # return error response
+            current_translation_type = translation.translation_type
         except Translation.DoesNotExist:
             return Response(
                 {
-                    "error": "No translation found for the given translation_id and target_lang."
+                    "error": "No translation found for the given translation_id and target_language."
                 },
                 status=status.HTTP_404_NOT_FOUND,
             )
 
-        # Return the appropriate response depending on whether a new translation was created or not
-        if created:
+        # If the translation belongs to the current user, update the translation
+        if translation.user == user:
+            translation.payload = payload
+            translation.save()
+            
             return Response(
-                {"message": "Translation created successfully."},
-                status=status.HTTP_201_CREATED,
-            )
+            {"message": "Translation updated successfully.", "id":translation.id, "data": translation.payload}, status=status.HTTP_200_OK)
 
-        return Response(
-            {"message": "Translation updated successfully."}, status=status.HTTP_200_OK
-        )
+        # else if the user is not assigned, create a new translation object
+        else: 
+
+            # Check if the translation doesn't have a user 
+            if translation.user is not None: 
+                return Response(
+                    {"error": "Translation already has a user assigned to it. You cannot edit this translation."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
 
+            new_translation = Translation.objects.create(
+                translation_type=MANUALLY_CREATED,
+                parent=translation,
```

Comment:
Here, the type should be `HUMAN_EDITED`.

Why? Because, `if translation.user is None`, it means that the parent is of `MACHINE_GENERATED` type. Hence the child should be `UPDATED_MACHINE_GENERATED`.

Reply:
I understand that these things could be confusing. Please feel free to let me know if you have any suggestions on how to make this more convenient for readers of the codebase to understand. One option could be adding extensive comments.

Reply:
Yeah, it can be a bit confusing. One way is to have a common document. This way, if the requirements change, we can change them in the source directly. For now, I am trying to leave comments wherever I can.